app/experimenter/nimbus-ui/src/components/PageRequestReview/index.test.tsx

```diff
@@ -215,18 +243,58 @@ describe("PageRequestReview", () => {
     await act(async () => void fireEvent.click(launchButton));
   });
 
-  it("handles submission with bad server data", async () => {
+  it("handles approval of launch as expected", async () => {
     const { mock, experiment } = mockExperimentQuery("demo-slug", {
-      status: NimbusExperimentStatus.DRAFT,
+      ...reviewRequestedBaseProps,
+      canReview: true,
     });
-    const mutationMock = createPublishStatusMutationMock(experiment.id!);
-    // @ts-ignore - intentionally breaking this type for error handling
-    delete mutationMock.result.data.updateExperiment;
-    render(<Subject mocks={[mock, mutationMock]} />);
-    await launchFromDraftToReview();
-    await waitFor(() =>
-      expect(screen.getByTestId("submit-error")).toBeInTheDocument(),
+    const mutationMock = createFullStatusMutationMock(
+      experiment.id!,
+      NimbusExperimentStatus.DRAFT,
+      NimbusExperimentPublishStatus.APPROVED,
     );
+    const triggerKintoMutationMock = mockTriggerCheckKintoPushQueueNowMutation();
+    render(<Subject mocks={[mock, mutationMock, triggerKintoMutationMock]} />);
+    const approveButton = await screen.findByTestId("approve-request");
+    fireEvent.click(approveButton);
+    const openRemoteSettingsButton = await screen.findByTestId(
+      "open-remote-settings",
+    );
+    fireEvent.click(openRemoteSettingsButton);
+    await waitFor(() => {
+      expect(mockWindowOpen).toHaveBeenCalledWith(
+        MOCK_CONFIG.kintoAdminUrl,
+        "_blank",
+      );
+    });
+  });
+
+  it("handles rejection of launch as expected", async () => {
+    const expectedReason = "This smells bad.";
+    const { mock, experiment } = mockExperimentQuery("demo-slug", {
+      ...reviewRequestedBaseProps,
+      canReview: true,
+    });
+    const mutationMock = createFullStatusMutationMock(
+      experiment.id!,
+      NimbusExperimentStatus.DRAFT,
+      NimbusExperimentPublishStatus.IDLE,
+      expectedReason,
+    );
+    render(<Subject mocks={[mock, mutationMock]} />);
+    const rejectButton = await screen.findByTestId("reject-request");
+    fireEvent.click(rejectButton);
+    const rejectSubmitButton = await screen.findByTestId("reject-submit");
+    const rejectReasonField = await screen.findByTestId("reject-reason");
+    expect(rejectSubmitButton).not.toBeEnabled();
+    fireEvent.change(rejectReasonField, {
+      target: { value: expectedReason },
+    });
+    fireEvent.blur(rejectReasonField);
```

Comment:
Yeah I thought we discussed this collectively and agreed not to do this?

Reply:
IIRC, we don't disable buttons so work in progress can be saved. But we *do* disable when the mutation itself can't be submitted because the required data is missing. Here, we don't get a usable server-side submission error if we allow submission - it's a GQL type error that we can't parse into something sensible

Reply:
I can poke at this a bit though and see if I can switch this to a client-side error

Reply:
Okay, I removed the button disabling logic. Also noticed a bug where the client-side validation message that *should* have been there wasn't showing up. So now, you should be able to click the Reject button and see the client-side message but not have the mutation attempted until the form is valid.